Spatial.gd

```diff
@@ -690,30 +700,35 @@ func gen_adjustment_nodes():
 				continue
 			var new_gizmo = gizmo_scene.instance()
 			new_gizmo.translation = gizmo_position
-			new_gizmo.link_point("path", route, path2d, i)
-			new_gizmo.connect("selected", self, "on_gizmo_selected")
+			new_gizmo.connect("selected", self, "on_gizmo_selected", ["path", [route, path2d], i])
 			new_gizmo.connect("deselected", self, "on_gizmo_deselected")
-			new_gizmo.connect("update", self, "on_update_made")
+			new_gizmo.connect("update", self, "on_point_updated")
 			$Gizmos.add_child(new_gizmo)
 	for icon in category3d.icons:
 		var new_gizmo = gizmo_scene.instance()
 		new_gizmo.translation = icon.translation
-		new_gizmo.link_point("icon", icon)
-		new_gizmo.connect("selected", self, "on_gizmo_selected")
+		new_gizmo.connect("selected", self, "on_gizmo_selected", ["icon", [icon]])
 		new_gizmo.connect("deselected", self, "on_gizmo_deselected")
-		new_gizmo.connect("update", self, "on_update_made")
+		new_gizmo.connect("update", self, "on_point_updated")
 		$Gizmos.add_child(new_gizmo)
 
 
 var currently_selected_node = null
-func on_gizmo_selected(object):
-	self.currently_selected_node = object
+func on_gizmo_selected(object, point_type: String, nodes: Array, point_index: int):
+	self.currently_selected_node = SelectedNode.new()
+	self.currently_selected_node.gizmo = object
+	self.currently_selected_node.point_type = point_type
 	$Control/Dialogs/NodeEditorDialog/ScrollContainer/VBoxContainer/DeleteNode.disabled = false
 	# Only enable these buttons if the object selected is a point on the path not an icon
-	if object.point_type == "path":
+	if point_type == "path":
```

Comment:
Can we have logic that does not rely on `point_type` branching in multiple places?